mne/viz/evoked.py

```diff
@@ -1565,15 +1582,34 @@ def _format_evokeds_colors(evokeds, cmap, colors):
             raise ValueError('If evokeds is a dict and a cmap is passed, '
                              'you must specify the colors.')
     for cond in evokeds.keys():
-        _validate_type(cond, 'str', "Conditions")
+        _validate_type(cond, str, "Conditions")
+
+    all_types = []
+    for condition in evokeds.values():
+        if isinstance(condition, (Epochs, Evoked)):
+            all_types.append(type(condition))
+        else:
+            for item in condition:
+                all_types.append(type(item))
+    all_types = set(all_types)
+    if len(all_types) != 1:
+        raise ValueError("All instances in `evokeds` must be of the same type "
+                         "(Evokeds OR Epochs), got " + str(all_types))
```

Comment:
please use/write a check function in check.py

also I am not super happy to see something called evokeds that can contains epochs. It should ne renamed to "inst"